### PR TITLE
chore(examples): upgrade examples dependencies

### PR DESCRIPTION
**Pull request by ruisaraiva19**
Changes:

- [x] Update examples dependencies
- [x] Move `@vitejs/plugin-react` and `vite` packages from `dependencies` to `devDependencies`
- [x] Update GitHub Actions to use Node 20 and pnpm 8 (use Node version from `.nvmrc`)
- [x] Remove some unused dependencies

### CODE REVIEW

**Comment by schiller-manuel** [comment]
can you please rebase?

**Comment by ruisaraiva19** [comment]
> can you please rebase?

yes, working on it!

**Comment by ruisaraiva19** [comment]
@schiller-manuel done ✅

File: examples/react/basic-file-based-codesplitting/src/routeTree.gen.ts [hunk 1]
```diff
@@ -1,134 +1,105 @@
...
 
-const PostsComponentImport = new FileRoute("/posts").createRoute()
-const LayoutLayoutBTestComponentImport = new FileRoute(
-  "/_layout/layout-b/test",
-).createRoute()
+const PostsComponentImport = new FileRoute('/posts').createRoute()
 
 const PostsComponentRoute = PostsComponentImport.update({
-  path: "/posts",
+  path: '/posts',
```
Comment (schiller-manuel) [inline examples/react/basic-file-based-codesplitting/src/routeTree.gen.ts:14]: why are the quotes changed? does this originate from https://github.com/TanStack/router/pull/948 ?

Comment (ruisaraiva19) [reply examples/react/basic-file-based-codesplitting/src/routeTree.gen.ts:14]: I just run `pnpm run prettier` to format all files

**Pull request merged**